app/src/main/java/com/rkm/tasky/feature/loginscreen/screen/LoginScreen.kt

```diff
@@ -0,0 +1,229 @@
+package com.rkm.tasky.feature.loginscreen.screen
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.wrapContentSize
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavHostController
+import com.rkm.tasky.R
+import com.rkm.tasky.feature.loginscreen.viewmodel.LoginViewModel
+import com.rkm.tasky.ui.component.PasswordTextField
+import com.rkm.tasky.ui.component.ValidationTextField
+import com.rkm.tasky.ui.theme.LoginSignUpTextColor
+
+@Composable
+fun LoginScreenRoot(
+    navHostController: NavHostController,
+    viewModel: LoginViewModel = hiltViewModel()
+) {
+
+    val email by viewModel.emailState.collectAsStateWithLifecycle()
+    val isValidEmail by viewModel.isValidEmailState.collectAsStateWithLifecycle()
+    val password by viewModel.passwordState.collectAsStateWithLifecycle()
+    val showPassword by viewModel.showPasswordState.collectAsStateWithLifecycle()
+
+    LoginScreen(
+        email = email,
+        password = password,
+        isValidEmail = isValidEmail,
+        showPassword = showPassword,
+        onEmailChange = viewModel::updateEmail,
+        onPasswordChange = viewModel::updatePassword,
+        onLoginButtonClick = { /*TODO*/ },
+        modifier = Modifier
+    )
```

Comment:
Could it not be simpler to pass a single state object here? This way, you wouldn’t have to update the signature of this composable each time you add some state property to that class. Your signature could look like this:

```
@Composable
fun LoginScreen(
  val state: LoginState,
  val onEmailChange: (String) -> Unit,
  val onPasswordChange: (String) -> Unit
) {
  Text(text = state.email)
}
```

Tip - you can similarly expose a single function to handle any interactions on your screen to pass to your viewmodel. With this approach you’d again not have to update the function signature when adding a new interaction (onclick) to your screen. _You’d also not break your `@Preview`functions each time you add a piece of state or another lambda._ You just need to find a way to model these actions 🤔